dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs

```diff
@@ -0,0 +1,254 @@
+            {
+                // The model replied with a message - check for final answer
+                var messageContent = (await chatResult.GetChatMessageAsync(cancellationToken).ConfigureAwait(false)).Content;
+
+                Match finalAnswerMatch = s_finalAnswerRegex.Match(messageContent);
+                if (finalAnswerMatch.Success)
+                {
+                    return new FunctionCallingStepwisePlannerResult
+                    {
+                        FinalAnswer = finalAnswerMatch.Groups[1].Value.Trim(),
+                        ChatHistory = chatHistoryForSteps,
+                        Iterations = i + 1,
+                    };
+                }
+
+                // No final answer found yet, so continue looping through steps
+                continue;
+            }
+
+            // Look up SKFunction
+            if (!this._kernel.Functions.TryGetFunctionAndContext(functionResponse, out ISKFunction? pluginFunction, out ContextVariables? funcContext))
+            {
+                throw new SKException($"Function {functionResponse.PluginName}.{functionResponse.FunctionName} not found in kernel.");
+            }
+
+            // Execute function and add to chat history
+            var result = (await this._kernel.RunAsync(funcContext, cancellationToken, pluginFunction).ConfigureAwait(false)).GetValue<object>();
+
+            if (result is RestApiOperationResponse apiResponse)
+            {
+                resultContent = apiResponse.Content as string ?? string.Empty;
+            }
+            else if (result is string str)
+            {
+                resultContent = str;
+            }
+            else
+            {
+                resultContent = JsonSerializer.Serialize(result);
+            }
+
+            chatHistoryForSteps.AddFunctionMessage(resultContent, functionResponse.FullyQualifiedName);
+        }
+
+        // We've completed the max iterations, but the model hasn't returned a final answer.
+        return new FunctionCallingStepwisePlannerResult
+        {
+            FinalAnswer = string.Empty,
+            ChatHistory = chatHistoryForSteps,
+            Iterations = this.Config.MaxIterations,
+        };
+    }
+
+    private async Task<IChatResult> GetCompletionWithFunctionsAsync(
+            ChatHistory chatHistory,
+            CancellationToken cancellationToken)
+    {
+        var requestSettings = this.PrepareOpenAIRequestSettingsWithFunctions();
+        return (await this._chatCompletion.GetChatCompletionsAsync(chatHistory, requestSettings, cancellationToken).ConfigureAwait(false))[0];
+    }
+
+    private async Task<string> GetFunctionsManualAsync(CancellationToken cancellationToken)
+    {
+        var availableFunctions = await this._kernel.Functions.GetFunctionsAsync(this.Config, null, this._logger, cancellationToken).ConfigureAwait(false);
+
+        return string.Join("\n\n", availableFunctions.Select(x => x.ToManualString()));
+    }
```

Comment:
This manual should match the format of the function descriptions sent to OpenAI - it currently does not.